reconcile/utils/terrascript_aws_client.py

```diff
@@ -5832,6 +5840,66 @@ def populate_tf_resource_secrets_manager(self, spec: ExternalResourceSpec) -> No
 
         self.add_resources(account, tf_resources)
 
+    @staticmethod
+    def _unflatten_dotted_keys_dict(flat_dict: dict[str, str]) -> dict[str, Any]:
+        """Convert a flat dictionary with dotted keys to a nested dictionary.
+
+        Example:
+            {"db.host": "localhost", "db.port": "5432"} ->
+            {"db": {"host": "localhost", "port": "5432"}}
+
+        Raises:
+            ValueError: If there are conflicting keys (e.g., "a.b" and "a.b.c")
+        """
+        result: dict[str, Any] = {}
+        for key, value in flat_dict.items():
+            parts = key.split(".")
+            current = result
+            for i, part in enumerate(parts[:-1]):
+                if part not in current:
+                    current[part] = {}
+                elif not isinstance(current[part], dict):
+                    # Conflict: trying to traverse through a non-dict value
+                    conflicting_path = ".".join(parts[: i + 1])
+                    raise ValueError(
+                        f"Conflicting keys detected: '{conflicting_path}' is both a "
+                        f"value and a nested path in key '{key}'"
+                    )
+                current = current[part]
+
+            # Check if we're trying to set a value where a dict already exists
+            if parts[-1] in current and isinstance(current[parts[-1]], dict):
+                raise ValueError(
+                    f"Conflicting keys detected: '{key}' conflicts with nested keys"
+                )
+
+            current[parts[-1]] = value
+
+        return result
+
+    @staticmethod
+    def _apply_secret_format(
+        secret_format: str, secret_data: dict[str, str]
+    ) -> dict[str, str]:
+        # Convert flat dict with dotted keys to nested dict for Jinja2
+        nested_secret_data = TerrascriptClient._unflatten_dotted_keys_dict(secret_data)
+        rendered_data = process_jinja2_template(secret_format, nested_secret_data)
```

Comment:
@rporres since this is a new feature, how about we put `secret_data` in a new var to avoid structure change?

Like this

```python
rendered_data = process_jinja2_template(secret_format, {"secret": secret_data})
```

so the jinja template is

```
{"password":"{{ secret["db.password"] }}","username":"{{ secret["db.user"] }}"}
```

Btw there is a similar feature [output_format](https://github.com/app-sre/qontract-schemas/blob/main/schemas/openshift/terraform-output-format-1.yml) which enforce result to be `dict` in schema, should we do the same? It will be something like this:

```yaml
    secret_format:
      password: '{{ secret["db.password"] }}'
      username: '{{ secret["db.username"] }}'
```